After executing the "create sharding binding table rules (t_ssdb_tb_view, t_ssdb_tb);" statement, an error will be reported when the next SQL statement is executed

## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
we find java version: java8, full_version=1.8.0_342, full_path=/home/peilq_sharding/bisheng-jdk1.8.0_342//bin/java
ShardingSphere-5.2.1-SNAPSHOT
Commit ID: dirty-f1cab653542d80b9a2b7a5cb4d39c3277862ecda
Commit Message: fix alter view rename bug. (#20979)
Branch: f1cab653542d80b9a2b7a5cb4d39c3277862ecda
Build time: 2022-09-16T14:31:19+0800

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy

### Expected behavior
After executing the "create sharding binding table rules (t_ssdb_tb_view, t_ssdb_tb);" statement, the next SQL statement is executed normally

### Actual behavior
After executing the "create sharding binding table rules (t_ssdb_tb_view, t_ssdb_tb);" statement, an error will be reported when the next SQL statement is executed

### Reason analyze (If you can)

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
```
drop view if exists t_ssdb_tb_view;
drop view if exists t_ssdb_tb_view_new;
drop table if exists t_ssdb_tb;
create table t_ssdb_tb(id int,c_id int,a int,b char(10));
insert into t_ssdb_tb values (1,1,10,'test11'),(1,2,10,'Test12'),(1,2,null,'test12'),
(2,1,20,'test21'),(2,2,20,'test22'),(2,3,null,'Test22'),
(3,3,null,'test3'),(3,3,null,'test3'),(4,4,40,'test4'),(5,5,50,'Test5');
show sharding binding table rules;
create sharding binding table rules (t_ssdb_tb_view, t_ssdb_tb);
select pg_sleep(2);
```

```
drop view if exists t_ssdb_tb_view;
DROP VIEW
drop view if exists t_ssdb_tb_view_new;
DROP VIEW
drop table if exists t_ssdb_tb;
DROP TABLE
create table t_ssdb_tb(id int,c_id int,a int,b char(10));
CREATE TABLE
insert into t_ssdb_tb values (1,1,10,'test11'),(1,2,10,'Test12'),(1,2,null,'test12'),
(2,1,20,'test21'),(2,2,20,'test22'),(2,3,null,'Test22'),
(3,3,null,'test3'),(3,3,null,'test3'),(4,4,40,'test4'),(5,5,50,'Test5');
INSERT 0 10

show sharding binding table rules;
 sharding_binding_tables
-------------------------
(0 rows)

create sharding binding table rules (t_ssdb_tb_view, t_ssdb_tb);
SUCCESS
select pg_sleep(2);
ERROR:  java.lang.NullPointerException
```

background log

```
[INFO ] 2022-09-16 15:39:04.601 [Connection-3-ThreadExecutor] ShardingSphere-SQL - Logic SQL: select pg_sleep(2);
[INFO ] 2022-09-16 15:39:04.601 [Connection-3-ThreadExecutor] ShardingSphere-SQL - SQLStatement: OpenGaussSelectStatement(super=SelectStatement(super=AbstractSQLStatement(parameterCount=0, parameterMarkerSegments=[], commentSegments=[]), projections=ProjectionsSegment(startIndex=7, stopIndex=17, projections=[ExpressionProjectionSegment(startIndex=7, stopIndex=17, text=pg_sleep(2), expr=FunctionSegment(startIndex=7, stopIndex=17, functionName=pg_sleep, parameters=[LiteralExpressionSegment(startIndex=16, stopIndex=16, literals=2)], text=pg_sleep(2), owner=null), alias=Optional.empty)], distinctRow=false), from=null, where=Optional.empty, groupBy=Optional.empty, having=Optional.empty, orderBy=Optional.empty, combines=[]), limit=Optional.empty, lock=Optional.empty, window=Optional.empty)
[INFO ] 2022-09-16 15:39:04.601 [Connection-3-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds1_1 ::: select pg_sleep(2);
[ERROR] 2022-09-16 15:39:04.601 [Connection-3-ThreadExecutor] o.a.s.p.f.c.CommandExecutorTask - Exception occur:
java.lang.NullPointerException: null
        at org.apache.shardingsphere.transaction.xa.XAShardingSphereTransactionManager.isInTransaction(XAShardingSphereTransactionManager.java:73)
        at org.apache.shardingsphere.proxy.backend.communication.jdbc.datasource.JDBCBackendDataSource.isInTransaction(JDBCBackendDataSource.java:118)
        at org.apache.shardingsphere.proxy.backend.communication.jdbc.datasource.JDBCBackendDataSource.createConnection(JDBCBackendDataSource.java:109)
        at org.apache.shardingsphere.proxy.backend.communication.jdbc.datasource.JDBCBackendDataSource.getConnections(JDBCBackendDataSource.java:80)
        at org.apache.shardingsphere.proxy.backend.communication.jdbc.datasource.JDBCBackendDataSource.getConnections(JDBCBackendDataSource.java:53)
        at org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.JDBCBackendConnection.createNewConnections(JDBCBackendConnection.java:96)
        at org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.JDBCBackendConnection.getConnections(JDBCBackendConnection.java:87)
        at org.apache.shardingsphere.infra.executor.sql.prepare.driver.DriverExecutionPrepareEngine.group(DriverExecutionPrepareEngine.java:88)
        at org.apache.shardingsphere.infra.executor.sql.prepare.AbstractExecutionPrepareEngine.prepare(AbstractExecutionPrepareEngine.java:62)
        at org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutor.useDriverToExecute(ProxySQLExecutor.java:187)
        at org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutor.execute(ProxySQLExecutor.java:153)
        at org.apache.shardingsphere.proxy.backend.communication.ProxySQLExecutor.execute(ProxySQLExecutor.java:147)
        at org.apache.shardingsphere.proxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.execute(JDBCDatabaseCommunicationEngine.java:130)
        at org.apache.shardingsphere.proxy.backend.handler.data.impl.UnicastDatabaseBackendHandler.execute(UnicastDatabaseBackendHandler.java:76)
        at org.apache.shardingsphere.proxy.frontend.opengauss.command.query.simple.OpenGaussComQueryExecutor.execute(OpenGaussComQueryExecutor.java:77)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:111)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:78)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:750)
```


### Example codes for reproduce this issue (such as a github link).

--- a/shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/rule/TransactionRule.java
+++ b/shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/rule/TransactionRule.java
@@ -57,7 +57,7 @@ public final class TransactionRule implements GlobalRule, ResourceHeldRule<Shard
     private volatile ShardingSphereTransactionManagerEngine resource;
     
     public TransactionRule(final TransactionRuleConfiguration ruleConfig, final Map<String, ShardingSphereDatabase> databases, final InstanceContext instanceContext) {
-        log.info("Create transaction rule");
+        log.debug("Create transaction rule");
         configuration = ruleConfig;
         defaultType = TransactionType.valueOf(ruleConfig.getDefaultType().toUpperCase());
         providerType = ruleConfig.getProviderType();
@@ -96,7 +96,7 @@ public final class TransactionRule implements GlobalRule, ResourceHeldRule<Shard
         if (null == database) {
             return;
         }
-        log.info("Transaction rule add resource: {}", database.getName());
+        log.debug("Transaction rule add resource: {}", database.getName());
         rebuildEngine();
     }
     
@@ -105,13 +105,13 @@ public final class TransactionRule implements GlobalRule, ResourceHeldRule<Shard
         if (!databases.containsKey(databaseName.toLowerCase())) {
             return;
         }
-        log.info("Transaction rule close resource: {}", databaseName);
+        log.debug("Transaction rule close resource: {}", databaseName);
         rebuildEngine();
     }
     
     @Override
     public synchronized void closeStaleResource() {
-        log.info("Transaction rule close all resources");
+        log.debug("Transaction rule close all resources");
         closeEngine();
     }
     

--- a/shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/rule/TransactionRule.java
+++ b/shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/rule/TransactionRule.java
@@ -57,6 +57,7 @@ public final class TransactionRule implements GlobalRule, ResourceHeldRule<Shard
     private volatile ShardingSphereTransactionManagerEngine resource;
     
     public TransactionRule(final TransactionRuleConfiguration ruleConfig, final Map<String, ShardingSphereDatabase> databases, final InstanceContext instanceContext) {
+        log.info("Create transaction rule");
         configuration = ruleConfig;
         defaultType = TransactionType.valueOf(ruleConfig.getDefaultType().toUpperCase());
         providerType = ruleConfig.getProviderType();
@@ -95,6 +96,7 @@ public final class TransactionRule implements GlobalRule, ResourceHeldRule<Shard
         if (null == database) {
             return;
         }
+        log.info("Transaction rule add resource: {}", database.getName());
         rebuildEngine();
     }
     
@@ -103,11 +105,13 @@ public final class TransactionRule implements GlobalRule, ResourceHeldRule<Shard
         if (!databases.containsKey(databaseName.toLowerCase())) {
             return;
         }
+        log.info("Transaction rule close resource: {}", databaseName);
         rebuildEngine();
     }
     
     @Override
     public synchronized void closeStaleResource() {
+        log.info("Transaction rule close all resources");
         closeEngine();
     }
     

--- a/shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/main/java/org/apache/shardingsphere/transaction/xa/XAShardingSphereTransactionManager.java
+++ b/shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/main/java/org/apache/shardingsphere/transaction/xa/XAShardingSphereTransactionManager.java
@@ -39,8 +39,6 @@ import java.sql.SQLException;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.TimeUnit;
 
 /**
  * ShardingSphere Transaction manager for XA.
@@ -51,15 +49,11 @@ public final class XAShardingSphereTransactionManager implements ShardingSphereT
     
     private XATransactionManagerProvider xaTransactionManagerProvider;
     
-    // TODO Add a global lock to prevent the transaction manager from being used when modifying the rule before it is created.
-    private final CountDownLatch latch = new CountDownLatch(1);
-    
     @Override
     public void init(final DatabaseType databaseType, final Collection<ResourceDataSource> resourceDataSources, final String providerType) {
         xaTransactionManagerProvider = XATransactionManagerProviderFactory.getInstance(providerType);
         xaTransactionManagerProvider.init();
         resourceDataSources.forEach(each -> cachedDataSources.put(each.getOriginalName(), newXATransactionDataSource(databaseType, each)));
-        latch.countDown();
     }
     
     private XATransactionDataSource newXATransactionDataSource(final DatabaseType databaseType, final ResourceDataSource resourceDataSource) {
@@ -73,13 +67,10 @@ public final class XAShardingSphereTransactionManager implements ShardingSphereT
         return TransactionType.XA;
     }
     
-    @SneakyThrows({SystemException.class, InterruptedException.class})
+    @SneakyThrows(SystemException.class)
     @Override
     public boolean isInTransaction() {
-        if (!latch.await(3000, TimeUnit.MILLISECONDS)) {
-            return false;
-        }
-        return Status.STATUS_NO_TRANSACTION != xaTransactionManagerProvider.getTransactionManager().getStatus();
+        return xaTransactionManagerProvider != null && Status.STATUS_NO_TRANSACTION != xaTransactionManagerProvider.getTransactionManager().getStatus();
     }
     
     @Override

--- a/shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/main/java/org/apache/shardingsphere/transaction/xa/XAShardingSphereTransactionManager.java
+++ b/shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/main/java/org/apache/shardingsphere/transaction/xa/XAShardingSphereTransactionManager.java
@@ -39,6 +39,8 @@ import java.sql.SQLException;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
 
 /**
  * ShardingSphere Transaction manager for XA.
@@ -49,11 +51,15 @@ public final class XAShardingSphereTransactionManager implements ShardingSphereT
     
     private XATransactionManagerProvider xaTransactionManagerProvider;
     
+    // TODO Add a global lock to prevent the transaction manager from being used when modifying the rule before it is created.
+    private final CountDownLatch latch = new CountDownLatch(1);
+    
     @Override
     public void init(final DatabaseType databaseType, final Collection<ResourceDataSource> resourceDataSources, final String providerType) {
         xaTransactionManagerProvider = XATransactionManagerProviderFactory.getInstance(providerType);
         xaTransactionManagerProvider.init();
         resourceDataSources.forEach(each -> cachedDataSources.put(each.getOriginalName(), newXATransactionDataSource(databaseType, each)));
+        latch.countDown();
     }
     
     private XATransactionDataSource newXATransactionDataSource(final DatabaseType databaseType, final ResourceDataSource resourceDataSource) {
@@ -67,9 +73,12 @@ public final class XAShardingSphereTransactionManager implements ShardingSphereT
         return TransactionType.XA;
     }
     
-    @SneakyThrows(SystemException.class)
+    @SneakyThrows({SystemException.class, InterruptedException.class})
     @Override
     public boolean isInTransaction() {
+        if (!latch.await(3000, TimeUnit.MILLISECONDS)) {
+            return false;
+        }
         return Status.STATUS_NO_TRANSACTION != xaTransactionManagerProvider.getTransactionManager().getStatus();
     }
     

